app/assets/javascripts/toggle_name_and_email_input.js

```diff
@@ -17,6 +17,10 @@
         {
           link.hide();
         }
+        else if ( targetScope === '[data-no-sunet-target]' )
+        {
+          $('form').find('input[type=text]').filter(':visible:first').focus();
```

Comment:
I feel like this is scoped too broadly.  If we were to add another form anywhere above this form in the DOM for any reason, this would unexpectedly attach focus to the first visible text input in **that** form.

Also, on line 10 there is a form variable (although I don't particularly like that either, because again, if we add a form after this one in the DOM we'll be scoped to the wrong form).

Here's a suggestion:

1. Change the variable in line 10 to use [`closest()`](https://api.jquery.com/closest/) to find the parent form.
2. Change this line to be scoped to finding inputs within that form.